sycl/include/sycl/ext/oneapi/experimental/annotated_ptr/annotated_ptr.hpp

```diff
@@ -151,17 +155,58 @@ class annotated_ref<T, detail::properties_t<Props...>> {
     return *this;
   }
 
-  PROPAGATE_OP(+=)
-  PROPAGATE_OP(-=)
-  PROPAGATE_OP(*=)
-  PROPAGATE_OP(/=)
-  PROPAGATE_OP(%=)
-  PROPAGATE_OP(^=)
-  PROPAGATE_OP(&=)
-  PROPAGATE_OP(|=)
+  this_t &operator++() {
+    T &t = annotated_load();
+    ++t;
+    return *this;
+  }
+
+  this_t &operator++(int) {
+    T &t = annotated_load();
+    t++;
+    return *this;
+  }
+
+  this_t &operator--() {
+    T &t = annotated_load();
+    --t;
+    return *this;
+  }
+
+  this_t &operator--(int) {
+    T &t = annotated_load();
+    t--;
+    return *this;
+  }
+
+#ifdef __SYCL_DEVICE_ONLY__
+#define PROPAGATE_BINARY_OP(op)                                                \
+  annotated_ref operator op##=(const T &rhs) {                                 \
+    (*__builtin_intel_sycl_ptr_annotation(                                     \
+        m_Ptr, detail::PropertyMetaInfo<Props>::name...,                       \
+        detail::PropertyMetaInfo<Props>::value...)) op## = rhs;                \
```

Comment:
Shouldn't we annotate both the load and store?

Even if this isn't needed, I would like us to specify and implement the compound operators using the non-compound operators so it is possible to implement on top of builtin load+store. We agreed in https://github.com/intel/llvm/pull/10124 that this is the correct solution.

BTW: You didn't do the equivalent update of the spec for this changes.